### PR TITLE
User merger patch 0

### PR DESCRIPTION
**Pull request by data-doge**
@mixmix ey yeeees thank you for your PR + thx for thinking through this problem -- i agree with your solution

here's a PR to your PR where i've:
  - merged your user merger service into `UserService`
  - some slight refactoring

curious to see what u think. if u like, plz merge into your branch and merge your branch into master 

thx again!

### CODE REVIEW

File: spec/services/user_service_spec.rb [hunk 1]
```diff
@@ -147,4 +147,98 @@
...
+      @bucket_to_transfer =       create(:bucket,     group: @group, user: @user_to_kill)
+      @comment_to_transfer =      create(:comment,                   user: @user_to_kill)
+    end
+
+    context "transfering memebership is going to double up membership for a group," do
+      before do
+        @existing_membership = create(:membership, member: @user_to_keep, group: @group)
+      end
+
+      context "they're both just normal memberships," do
```
Comment (data-doge) [inline spec/services/user_service_spec.rb]: here i changed 'normal users' to 'normal memberships'

File: app/services/user_service.rb [hunk 2]
```diff
@@ -23,4 +23,24 @@ def self.fetch_recent_activity_for(user:)
...
+      memberships = Membership.where(group: membership.group, member: [user_to_keep, user_to_kill])
+
+      if existing_membership = memberships.find_by(member: user_to_keep)
+        existing_membership.update(is_admin: true) if memberships.where(is_admin: true).any?
+        existing_membership.reactivate! if memberships.active.any?
+        membership.destroy
+      else
+        membership.update(member: user_to_keep)
+      end
+    end
```
Comment (data-doge) [inline app/services/user_service.rb]: i'm a ruby one-liner fanboy -- i can't help myself

File: spec/services/user_service_spec.rb (hunk 1 above)
Comment (mixmix) [inline spec/services/user_service_spec.rb]: nice and clear :)

File: app/services/user_service.rb (hunk 2 above)
Comment (mixmix) [inline app/services/user_service.rb]: oooo, that's tidy, I like it, nice work !

**Comment by mixmix** [comment]
This looks good to me @data-doge 
The main reason I put it in its own service was so that it didn't get casually used. I trust the named variable `user_to_kill` will invoke the right sort of thinking though ... 

merging this. and the other one.
I'll try a deploy and maybe try this out on silvia, then derek, then some enspiral users

**Comment by data-doge** [comment]
thank u !!!!!

**Comment by derekrazo** [comment]
Awesome work guys :)
ᐧ

**Pull request merged**